dotCMS/src/main/java/com/dotcms/rest/api/v1/folder/FolderHelper.java

```diff
@@ -98,4 +100,56 @@ public Folder loadFolderByURI(String siteName, User user, String uri) throws Dot
         return ret;
     }
 
+    /**
+     *
+     * @param hostId hostId where the folder lives
+     * @param folder parent folder to find
+     * @param user user making the request
+     * @return FolderView with the info of the folder requested and the subFolders
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    public FolderView loadFolderAndSubFoldersByPath(final String hostId, final String folder, final User user)
+            throws DotSecurityException, DotDataException {
+        final String uriParam = !folder.startsWith(StringPool.FORWARD_SLASH) ? StringPool.FORWARD_SLASH.concat(folder) : folder;
+        final Host host = APILocator.getHostAPI().find(hostId,user,false);
+        final Folder folderByPath = APILocator.getFolderAPI().findFolderByPath(uriParam, host, user, false);
+        if(!UtilMethods.isSet(host)) {
+            throw new IllegalArgumentException(String.format(" Couldn't find any host with id `%s` ",hostId));
+        }
+        if(!UtilMethods.isSet(folderByPath) || !UtilMethods.isSet(folderByPath.getInode())) {
+            throw new IllegalArgumentException(String.format(" Couldn't find any folder with name `%s` in the host `%s`",folder,hostId));
+        }
+
+        return getFolders(folderByPath,user);
+    }
+
+    /**
+     * This method returns a folder structure with their children recursively based on
+     * the folder returned by findFolderByPath
+     *
+     * @param folder  parent folder to  find
+     * @param user user
+     * @return FolderView a folder structure with their children recursively
+     */
+    private final FolderView getFolders(final Folder folder, final User user){
+
+        final List<FolderView> foldersChildCustoms = new LinkedList<>();
+        List<Folder> children = null;
+        try {
+            children = APILocator.getFolderAPI().findSubFolders(folder, user, false);
+        } catch (Exception e) {
```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5916479)